File: app/src/main/java/com/danielmaia/desafioandroid/service/RetrofitApi.java
package com.danielmaia.desafioandroid.service;

import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class RetrofitApi {
    private static Retrofit retrofit;
    private static final String BASE_URL = "https://api.github.com/";

    public static Retrofit getRetrofitInstance() {
        if (retrofit == null) {
            retrofit = new retrofit2.Retrofit.Builder()
                    .baseUrl(BASE_URL)
                    .addConverterFactory(GsonConverterFactory.create())
                    .build();
        }
        return retrofit;
    }
}

File: app/src/main/java/br/com/rmso/popularrepositories/retrofit/RetrofitAPI.kt
package br.com.rmso.popularrepositories.retrofit

import br.com.rmso.popularrepositories.retrofit.services.PullRequestService
import br.com.rmso.popularrepositories.retrofit.services.RepositoryService
import br.com.rmso.popularrepositories.utils.Constants
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

class RetrofitAPI {
    private val constants = Constants()

    private val retrofit: Retrofit = Retrofit.Builder()
        .baseUrl(constants.baseUrl)
        .addConverterFactory(GsonConverterFactory.create())
        .build()

    val repositoryService: RepositoryService
        get() = this.retrofit.create<RepositoryService>(
            RepositoryService::class.java)


    val pullRequestService: PullRequestService
        get() = this.retrofit.create<PullRequestService>(
            PullRequestService::class.java)
}